sdks/python/apache_beam/io/gcp/gcsio.py

```diff
@@ -458,7 +460,11 @@ def last_updated(self, path):
     request = storage.StorageObjectsGetRequest(
         bucket=bucket, object=object_path)
     datetime = self.client.objects.Get(request).updated
-    return (time.mktime(datetime.timetuple()) - time.timezone
+
+    utc_datetime = pytz.utc.localize(datetime)
+    utc_offset = utc_datetime.astimezone(tzlocal.get_localzone()).utcoffset().total_seconds()
```

Comment:
What do you think of explicitly specifying the beginning of the unix epoch:
```py
    seconds = (utc_datetime - datetime.datetime(1970, 1, 1, tzinfo=pytz.utc)).total_seconds()
```
(based on this answer: https://stackoverflow.com/a/17257177/63302)